botcore/_bot.py

```diff
@@ -104,8 +106,10 @@ def _connect_statsd(
         self.closing_tasks: list[asyncio.Task] = []
 
     async def load_extensions(self, module: types.ModuleType) -> None:
-        """Load all the extensions within the given module."""
-        for extension in walk_extensions(module):
+        """Load all the extensions within the given module and save them to ``self.all_extensions``."""
+        self.all_extensions = walk_extensions(module)
```

Comment:
Any ideas for how to document this attribute (and possibly other attributes)?

Reply:
It seems like it _should_ be possible by using `Attributes` in the class/init docstring, adding a docstring after the attr, or using `#:` on the line before it.

However all of these methods result in the follow error when building the docs `Handler <function doctree_read at 0x0000026AE96A80D0> for event 'doctree-read' threw an exception (exception: type object 'BotBase' has no attribute 'all_extensions')` I've spent about an hour googling around trying to figure out why this happens but haven't managed to get anywhere with it.

Reply:
This can be addressed later if there are concerns of holding up this PR.